src/main/scala/io/iohk/ethereum/transactions/PendingTransactionsManager.scala

```diff
@@ -21,29 +27,63 @@ object PendingTransactionsManager {
 class PendingTransactionsManager(peerManager: ActorRef, peerMessageBus: ActorRef) extends Actor {
 
   import PendingTransactionsManager._
+  import akka.pattern.ask
 
+  /**
+    * stores all pending transactions
+   */
   var pendingTransactions: Seq[SignedTransaction] = Nil
 
+  /**
+    * stores information which tx hashes are "known" by which peers
+    */
+  var knownTransactions: Map[ByteString, Set[PeerId]] = Map.empty
```

Comment:
Geth seems to keep at most a certain number of txs per peer https://github.com/ethereumproject/go-ethereum/blob/master/eth/peer.go#L42, should we also incorporate this? This isn't a blocker for this PR getting merged and could be a future optimization.